internal/workflow/preprocessing.go

```diff
@@ -221,14 +226,13 @@ func (w *PreprocessingWorkflow) Execute(
 			result.systemError(ctx, e, ev, "Unbagging the SIP has failed")
 			return result, nil
 		}
-		ev.Succeed(ctx, "SIP unbagged")
-
 		localPath = unbagResult.Path
-		rp, err := filepath.Rel(w.sharedPath, localPath)
-		if err != nil {
-			return nil, fmt.Errorf("Invalid path")
+		result.RelativePath, e = filepath.Rel(w.sharedPath, localPath)
+		if e != nil {
+			result.systemError(ctx, e, ev, "Unbagging the SIP has failed")
+			return result, nil
 		}
-		result.RelativePath = rp
+		ev.Succeed(ctx, "SIP unbagged")
```

Comment:
I was testing only Bags so this relative path update was hiding the error. I've changed this too, so it reports the possible error in the related event and the result.